### DISCUSSION TITLE
When using nuclei as golang libaray, An error has occurred

### DISCUSSION THREAD

**Discussion by Hu3sky** [projectdiscovery/nuclei#5310] [Q&A]
When I use the official demo：

```go
package main

import (
	"sync"
	"testing"

	nuclei "github.com/projectdiscovery/nuclei/v3/lib"
)

func TestAdd(t *testing.T) {
	// create nuclei engine with options
	ne, err := nuclei.NewThreadSafeNucleiEngine()
	if err != nil {
		panic(err)
	}
	// setup waitgroup to handle concurrency
	wg := &sync.WaitGroup{}

	// scan 1 = run dns templates on scanme.sh
	wg.Add(1)
	go func() {
		defer wg.Done()
		err = ne.ExecuteNucleiWithOpts([]string{"127.0.0.1:7001"}, nuclei.WithTemplateFilters(nuclei.TemplateFilters{ProtocolTypes: "http"}))
		if err != nil {
			panic(err)
		}
	}()

	// scan 2 = run http templates on honey.scanme.sh
	wg.Add(1)
	go func() {
		defer wg.Done()
		err = ne.ExecuteNucleiWithOpts([]string{"127.0.0.1:7001"}, nuclei.WithTemplateFilters(nuclei.TemplateFilters{Severity: "critical"}))
		if err != nil {
			panic(err)
		}
	}()

	// wait for all scans to finish
	wg.Wait()
	defer ne.Close()
}

```

panic:

# github.com/projectdiscovery/interactsh/pkg/server/acme
/Users/hu3sky/go/pkg/mod/github.com/projectdiscovery/interactsh@v1.2.0/pkg/server/acme/acme_certbot.go:39:3: unknown field DNSProvider in struct literal of type certmagic.DNS01Solver
/Users/hu3sky/go/pkg/mod/github.com/projectdiscovery/interactsh@v1.2.0/pkg/server/acme/acme_certbot.go:40:3: unknown field Resolvers in struct literal of type certmagic.DNS01Solver
# github.com/projectdiscovery/nuclei/v3/pkg/input/formats/openapi
/Users/hu3sky/go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.9/pkg/input/formats/openapi/examples.go:178:22: invalid operation: schema.Type == "boolean" (mismatched types *openapi3.Types and untyped string)
/Users/hu3sky/go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.9/pkg/input/formats/openapi/examples.go:180:22: invalid operation: schema.Type == "number" (mismatched types *openapi3.Types and untyped string)
/Users/hu3sky/go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.9/pkg/input/formats/openapi/examples.go:180:47: invalid operation: schema.Type == "integer" (mismatched types *openapi3.Types and untyped string)
/Users/hu3sky/go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.9/pkg/input/formats/openapi/examples.go:211:21: invalid operation: schema.Type == "integer" (mismatched types *openapi3.Types and untyped string)
/Users/hu3sky/go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.9/pkg/input/formats/openapi/examples.go:215:22: invalid operation: schema.Type == "string" (mismatched types *openapi3.Types and untyped string)
/Users/hu3sky/go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.9/pkg/input/formats/openapi/examples.go:229:22: invalid operation: schema.Type == "array" (mismatched types *openapi3.Types and untyped string)
/Users/hu3sky/go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.9/pkg/input/formats/openapi/examples.go:245:22: invalid operation: schema.Type == "object" (mismatched types *openapi3.Types and untyped string)
FAIL    catScanWeb/cmd/mainapp [build failed]

how to fix it

**Comment by cn-kali-team** [comment]
go get -u github.com/getkin/kin-openapi@v0.123.0
go get -u github.com/caddyserver/certmagic@v0.20.0

**Reply by blockisec** [reply]
after running the commands from @cn-kali-team the issue is gone. However, the official demo still does not work:

```
# github.com/projectdiscovery/nuclei/v3/pkg/reporting/trackers/gitlab
../../_gopath/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.3.0/pkg/reporting/trackers/gitlab/gitlab.go:121:16: cannot use &customLabels (value of type *"github.com/xanzy/go-gitlab".Labels) as *"github.com/xanzy/go-gitlab".LabelOptions value in struct literal

```

Edit: Update the `gitlab.go` file to this line `customLabels := gitlab.LabelOptions(labels)`.

**Reply by ogow** [reply]
@blockisec  have you been able to get past this somehow?, i get the same error...
```
cannot use &customLabels (value of type *"github.com/xanzy/go-gitlab".Labels) as *"github.com/xanzy/go-gitlab".LabelOptions value in struct literal
```

**Comment by dogancanbakir** [comment]
I couldn't repro the issue, make sure you have all the dependencies installed:

```console
$ go run .
[INF] Using Interactsh Server: oast.site
[0:00:05] | Templates: 232 | Hosts: 1 | RPS: 11 | Matched: 0 | Errors: 0 | Requests: 62/334 (18%)
[0:00:10] | Templates: 232 | Hosts: 1 | RPS: 12 | Matched: 0 | Errors: 0 | Requests: 130/334 (38%)
[CVE-2016-1555] http://honey.scanme.sh/boardDataWW.php
[CVE-2018-16167] http://honey.scanme.sh/upload
[CVE-2017-9506] http://honey.scanme.sh/plugins/servlet/oauth/users/icon-uri?consumerUri=http://cugfqi2hc8a7tip93urgkkff3yd6khzg4.oast.site
[CVE-2019-3929] http://honey.scanme.sh/cgi-bin/file_transfer.cgi
[CVE-2019-8451] http://honey.scanme.sh/plugins/servlet/gadgets/makeRequest
[0:00:15] | Templates: 232 | Hosts: 1 | RPS: 13 | Matched: 5 | Errors: 3 | Requests: 207/334 (61%)
...
```

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
